src/main/java/service/impl/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package service.impl;
+
+import dao.StorageDao;
+import dao.StorageDaoImpl;
+import java.util.List;
+import model.FruitTransaction;
+import service.FruitShopService;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private final StorageDao storageDao;
+    private final OperationStrategy operationStrategy;
+
+    public FruitShopServiceImpl(OperationStrategy optionalStrategy) {
+        this.operationStrategy = optionalStrategy;
+        this.storageDao = new StorageDaoImpl();
```

Comment:
let's also pass this dao using a constructor parameter instead of creating a new instance here directly. This will make your code more consistent